base/auspice_export.py

```diff
@@ -39,6 +39,23 @@ def process_freqs(freq):
     else:
         process.log.notify("Cannot export frequencies - pivots do not exist")
 
+def export_tip_frequency_json(process, prefix, indent):
+    def process_freqs(freq): # 6dp here, 4 above
+        return [round(x,6) for x in freq]
+
+    if not (hasattr(process, 'pivots') and hasattr(process, 'tree_frequencies')):
+        process.log.notify("Cannot export tip frequencies - pivots and/or tree_frequencies do not exist")
+        return;
+
+    freq_json = {'pivots':process_freqs(process.pivots)}
+
+    clade_tip_map = {n.clade: n.name for n in process.tree.tree.get_terminals()}
+
+    for clade, freq in process.tree_frequencies["global"].iteritems():
```

Comment:
Instead of looping through all the clades here, you can just loop through the tips and use `tip.clade` as the key to get the corresponding frequencies from `process.tree_frequencies["global"][tip.clade]`.

Reply:
038748012656420d39d932e5e4926cdb0b6db6b9